src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs

```diff
@@ -40,20 +40,29 @@ public class ExtendedAzureResourceManagerClient : IExtendedAzureResourceManagerC
         /// </summary>
         private const string DependencyName = "ARM";
 
+        /// <summary>
+        /// The HTTP request timeout for ARM calls, in minutes
+        /// </summary>
+        private const int HttpRequestTimeoutInMinutes = 5;
+
         private static readonly ConcurrentDictionary<string, ProviderInner> ProvidersCache = new ConcurrentDictionary<string, ProviderInner>(StringComparer.CurrentCultureIgnoreCase);
 
         private readonly ServiceClientCredentials credentials;
         private readonly IExtendedTracer tracer;
         private readonly Policy retryPolicy;
+        private readonly Policy<HttpResponseMessage> httpRetryPolicy;
         private readonly Uri baseUri;
+        private readonly IHttpClientWrapper httpClientWrapper;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ExtendedAzureResourceManagerClient"/> class
         /// </summary>
+        /// <param name="httpClientWrapper">The HTTP client wrapper</param>
         /// <param name="credentialsFactory">The credentials factory</param>
         /// <param name="tracer">The tracer</param>
-        public ExtendedAzureResourceManagerClient(ICredentialsFactory credentialsFactory, IExtendedTracer tracer)
+        public ExtendedAzureResourceManagerClient(IHttpClientWrapper httpClientWrapper, ICredentialsFactory credentialsFactory, IExtendedTracer tracer)
         {
+            this.httpClientWrapper = Diagnostics.EnsureArgumentNotNull(() => httpClientWrapper);
```

Comment:
>httpClientWrapper [](start = 17, length = 17)

why not use this.GetResourceManagementClient(subscriptionId).HttpClient?

Reply:
Possible. But doing it this way makes it easier to unit test this method. I prefer to keep it this way

---
In reply to: [232698744](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/19#discussion_r232698744) [](ancestors = 232698744)